src/Native/Runtime/amd64/StubDispatch.S

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+.intel_syntax noprefix
+#include <unixasmmacros.inc>
+
+// Macro that generates a stub consuming a cache with the given number of entries.
+.macro DEFINE_INTERFACE_DISPATCH_STUB entries
+
+LEAF_ENTRY RhpInterfaceDispatch\entries, _TEXT
+
+        // r10 currently contains the indirection cell address.
+        // load r11 to point to the cache block.
+        mov     r11, [r10 + OFFSETOF__InterfaceDispatchCell__m_pCache]
+
+        // Load the EEType from the object instance in rdi.
+        mov     rax, [rdi]
+
+        // Point r11 at the address of the first cache entry
+        // Switch to ATT syntax because add r11, OFFSETOF__InterfaceDispatchCache__m_rgEntries
+        // refuses to treat symbols as immediate values.
+        .att_syntax
+        addq     $OFFSETOF__InterfaceDispatchCache__m_rgEntries, %r11
+        .intel_syntax noprefix
+
+        // For each entry in the cache, see if its EEType type matches the EEType in rax.
+        // If so, call the second cache entry.  If not, skip the InterfaceDispatchCacheEntry.
+        .rept \entries
+            cmp     rax, [r11]
+            je      DispatchCall\entries
+
+            .att_syntax
+            addq    $SIZEOF__InterfaceDispatchCacheEntry, %r11
+            .intel_syntax noprefix
+        .endr
+
+        // r10 still contains the the indirection cell address.
+
+        jmp     C_FUNC(RhpInterfaceDispatchSlow)
+DispatchCall\entries:
+        add     r11, 0x8h
```

Comment:
Any specific reason to add 8h here as opposed to doing [r11+8] in the next instruction? Is r11 expected to be incremented?